### PR TITLE
add(pipe): pool commands template

### PR DESCRIPTION
**Pull request by Shenglai**
Add a pool commands template that I'm using for muse and mutect2.
https://github.com/NCI-GDC/muse-tool/blob/master/muse-tool/tools/muse_call.py#L26

Any thoughts?

### CODE REVIEW

File: cdis_pipe_utils/pipe_util.py [hunk 1]
```diff
@@ -120,3 +120,22 @@ def get_param(args, param_name):
...
+        df['cmd'] = cmd
+        unique_key_dict = {'uuid': uuid, 'cmd': cmd}
+        table_name = ('time_mem_%s' % step)
+        df_util.save_df_to_sqlalchemy(df, unique_key_dict, table_name, engine, logger)
+        logger.info('completed muse chunk call: %s' % str(cmd))
+    return output.wait()
+
+def multi_commands(uuid, cmds, step, thread_count, engine, logger):
+    pool = Pool(int(thread_count))
+    output = pool.map(partial(do_pool_commands, uuid=uuid, step=step, engine=engine, logger=logger), cmds)
```
Comment (MurphyMarkW) [inline cdis_pipe_utils/pipe_util.py]: Didn't think about using partial application. That's spiffy.

File: cdis_pipe_utils/pipe_util.py [hunk 2]
```diff
@@ -120,3 +120,22 @@ def get_param(args, param_name):
...
+    output = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    output_stdout = output.communicate()[1]
+    with lock:
+        logger.info('contents of output=%s' % output_stdout.decode().format())
+        df = time_util.store_time(uuid, cmd, output_stdout, logger)
+        df['cmd'] = cmd
+        unique_key_dict = {'uuid': uuid, 'cmd': cmd}
+        table_name = ('time_mem_%s' % step)
+        df_util.save_df_to_sqlalchemy(df, unique_key_dict, table_name, engine, logger)
+        logger.info('completed muse chunk call: %s' % str(cmd))
```
Comment (MurphyMarkW) [inline cdis_pipe_utils/pipe_util.py]: Prolly want to remove the hardcoded muse reference - maybe just declare the opposite of the first info log?
```python
logging.info('completed %s call: %s' % (step, cmd))
```

**Comment by MurphyMarkW** [comment]
I think you need a couple of imports for `Lock` and `Pool`.

So if I follow correctly, this is meant to repeat the same `step` with multiple, potentially different `cmd`s. It would be for when you have a block of commands that are logically grouped together, but could be executed asynchronously, right?

**Comment by Shenglai** [comment]
Sorry, I forget importing lock and pool, and yes, you are all right. I haven't tested 'step', but think it would work as well.

**Comment by Shenglai** [comment]
So, what I should do next? Delete this PR first, and then create a new PR with adding import lock and pool?

**Comment by MurphyMarkW** [comment]
Nope, just check out the `chore/multiprocess-pool` branch, add those changes, commit and push it. You'll see the changes appear here, and we'll keep reviewing.

**Comment by Shenglai** [comment]
Gonna close this legacy PR.

**Pull request closed without merging**